PWGCF/MultiparticleCorrelations/Tasks/O2AnalysisTaskThreePartCorr.cxx

```diff
@@ -0,0 +1,299 @@
+        QARegistry.fill(HIST("hTrackPhi"), track.phi());
+        if (TrackPID(track).first == track::PID::Pion) {
+          QARegistry.fill(HIST("hNSigmaPion"), track.pt(), track.tpcNSigmaPi());
+        } else if (TrackPID(track).first == track::PID::Kaon) {
+          QARegistry.fill(HIST("hNSigmaKaon"), track.pt(), track.tpcNSigmaKa());
+        } else if (TrackPID(track).first == track::PID::Proton) {
+          QARegistry.fill(HIST("hNSigmaProton"), track.pt(), track.tpcNSigmaPr());
+        }
+      }
+    }
+    // End of the Track QA
+
+    // Start of the V0-Track Correlations
+    for (const auto& trigger : v0s) {
+      if (V0Filters(trigger)) {
+
+        TSign = V0Sign(trigger);
+        if (TSign == 1) {
+          CandMass = trigger.mLambda();
+          QARegistry.fill(HIST("hInvMassLambda"), trigger.mLambda(), trigger.pt(), collision.centFT0C());
+        } else if (TSign == -1) {
+          CandMass = trigger.mAntiLambda();
+          QARegistry.fill(HIST("hInvMassAntiLambda"), trigger.mAntiLambda(), trigger.pt(), collision.centFT0C());
+        }
+
+        for (const auto& associate : tracks) {
+          if (TrackFilters(trigger, associate)) {
+
+            DeltaPhi = PhiCorrection(trigger.phi(), associate.phi());
+            DeltaEta = trigger.eta() - associate.eta();
+
+            if (CandMass >= 1.10 && CandMass <= 1.13) {
+              if (TrackPID(associate).first == track::PID::Pion) {
+                SECorrRegistry.fill(HIST("hSameLambdaPion_SGNL"), DeltaPhi, DeltaEta, collision.centFT0C(), collision.posZ(), trigger.pt(), associate.pt(), TSign, associate.sign());
+              } else if (TrackPID(associate).first == track::PID::Kaon) {
+                SECorrRegistry.fill(HIST("hSameLambdaKaon_SGNL"), DeltaPhi, DeltaEta, collision.centFT0C(), collision.posZ(), trigger.pt(), associate.pt(), TSign, associate.sign());
+              } else if (TrackPID(associate).first == track::PID::Proton) {
+                SECorrRegistry.fill(HIST("hSameLambdaProton_SGNL"), DeltaPhi, DeltaEta, collision.centFT0C(), collision.posZ(), trigger.pt(), associate.pt(), TSign, associate.sign());
+              }
+            } else {
+              if (TrackPID(associate).first == track::PID::Pion) {
+                SECorrRegistry.fill(HIST("hSameLambdaPion_SB"), DeltaPhi, DeltaEta, collision.centFT0C(), collision.posZ(), trigger.pt(), associate.pt(), TSign, associate.sign());
+              } else if (TrackPID(associate).first == track::PID::Kaon) {
+                SECorrRegistry.fill(HIST("hSameLambdaKaon_SB"), DeltaPhi, DeltaEta, collision.centFT0C(), collision.posZ(), trigger.pt(), associate.pt(), TSign, associate.sign());
+              } else if (TrackPID(associate).first == track::PID::Proton) {
+                SECorrRegistry.fill(HIST("hSameLambdaProton_SB"), DeltaPhi, DeltaEta, collision.centFT0C(), collision.posZ(), trigger.pt(), associate.pt(), TSign, associate.sign());
+              }
```

Comment:
Be aware that in the worst case this implies four conditional checks in the most inner loop
If there is excessive execution time this will be a candidate to improve